### PR TITLE
(BKR-1472) Enhancements to Contributor Guidance

### PR DESCRIPTION
**Pull request by Dakta**
Enhances directions for project contributors, including detailed installation instructions and process guidelines as well as Markdown formatting conventions.

Split from #1513

[skip ci]

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
 * Make sure you have a [GitHub account](https://github.com/signup/free)
-* Submit a ticket for your issue, assuming one does not already exist.
+* Submit a ticket for your issue on Jira, assuming one does not already exist.
   * Clearly describe the issue including steps to reproduce when it is a bug.
-  * File a ticket in the [BKR project](https://tickets.puppetlabs.com/issues/?jql=project%20%3D%20BKR)
+  * File the ticket in the [BKR project](https://tickets.puppetlabs.com/issues/?jql=project%20%3D%20BKR)
 * Fork the [Beaker repository on GitHub](https://github.com/puppetlabs/beaker)
 
-## Making Changes
+## Installation/Setup for Development
```
Comment (kevpl) [inline CONTRIBUTING.md]: please move installation to the installation doc. This goes for the "Recommended Tools" section above as well.

The contributor guide should be a resource for people to come back to over & over again as they go through their time contributing to beaker, so we shouldn't have one-time things here.

With that in mind, the Getting Started section should really be about Filing Bugs & the one-time JIRA creation steps can be moved to the ticket process doc.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
+  ```console
+    $ rake test:spec # assuming you have rbenv-bundler
+    # or
+    $ bundle exec rake test:spec # if you're *sure* your dependencies are tidy
+  ```
+
+## Making Changes
+
+* Create a topic branch from your fork of [puppetlabs/beaker](https://github.com/puppetlabs/beaker).
+  * Please title the branch after the beaker ticket you intend to address, i.e. `BKR-1234`.
```
Comment (kevpl) [inline CONTRIBUTING.md]: never thought the branch mattered. Did you end up running that test to confirm it does?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -30,9 +55,12 @@
     from our issue tracker.  The body describes the behavior without the patch,
     why this is a problem, and how the patch fixes the problem when applied.
 ````
-
 * During the time that you are working on your patch the master Beaker branch may have changed - you'll want to [rebase](http://git-scm.com/book/en/Git-Branching-Rebasing) on top of [Beaker's](https://github.com/puppetlabs/beaker) master branch before you submit your PR.  A successful rebase ensures that your PR will cleanly merge into Beaker.
 
+#### Courtesy
+
+Please do not introduce personal ignores into the `.gitignore`, such as IDE configurations, editor version files, or personal testing detritus. You may find it valuable to add the first two to [a global ignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore), and the third to [a repository-level ignore](https://help.github.com/articles/ignoring-files/#explicit-repository-excludes).
```
Comment (kevpl) [inline CONTRIBUTING.md]: let's not call what people who are trying to help us make 'detritus'. Something a little more neutral... artifacts?

File: CONTRIBUTING.md (hunk 2 above)
Comment (Dakta) [reply CONTRIBUTING.md]: Branch naming has meaning, but the functionality is actually broken in the integration. Basically, the name of the branch gets picked up and auto-linked into the Jira ticket as a "Branch", but doesn't understand which repository that branch is connected to. See attached and note that the branch attempts to link to the main repository, where no such branch exists.
image

File: CONTRIBUTING.md (hunk 3 above)
Comment (Dakta) [reply CONTRIBUTING.md]: No malice intended; "artifact" is definitely better.

**Pull request merged**